### ISSUE TITLE
Using getNotificationAppLaunchDetails instead of onDidReceiveBackgroundNotificationResponse

### ISSUE THREAD

**Issue by mehmetext** [MaikuB/flutter_local_notifications#1899]
**Describe the bug**
I wanted to use `onDidReceiveBackgroundNotificationResponse` but this function is never triggered. I trigger the function `getNotificationAppLaunchDetails` when I initialize the plugin instead of it. So, do you suggest me to use `getNotificationAppLaunchDetails` instead of `onDidReceiveBackgroundNotificationResponse`? Or why does not `onDidReceiveBackgroundNotificationResponse` trigger?

I imported correctly, I can send notification etc. but I can't handle the function `onDidReceiveBackgroundNotificationResponse` for some reason.

My NotificationService codes are below:
```Dart
import 'dart:convert';

import 'package:flutter_local_notifications/flutter_local_notifications.dart';
import 'package:local_notification_denemeler/utils/clg.dart';

@pragma('vm:entry-point')
void notificationTapBackground(NotificationResponse notificationResponse) {
  clg("notificationTapBackground()");

  NotificationService.onNotificationTapped(notificationResponse);
}

class NotificationService {
  static FlutterLocalNotificationsPlugin localNotifications =
      FlutterLocalNotificationsPlugin();

  static Future<void> init() async {
    clg("NotificationService.init()");

    localNotifications
        .resolvePlatformSpecificImplementation<
            AndroidFlutterLocalNotificationsPlugin>()
        ?.requestPermission();

    localNotifications.initialize(
      InitializationSettings(
        android: AndroidInitializationSettings("app_icon"),
      ),
      onDidReceiveBackgroundNotificationResponse: notificationTapBackground,
      onDidReceiveNotificationResponse: onNotificationTapped,
    );
  }

  static Future<void> checkNotificationClicked() async {
    clg("NotificationService.checkNotificationClicked()");
    NotificationAppLaunchDetails? details =
        await localNotifications.getNotificationAppLaunchDetails();

    if (details != null &&
        details.didNotificationLaunchApp &&
        details.notificationResponse != null) {
      onNotificationTapped(details.notificationResponse!);
    }
  }

  static Future<void> showSimpleNotification() async {
    clg("NotificationService.showSimpleNotification()");

    AndroidNotificationDetails androidNotificationDetails =
        AndroidNotificationDetails(
      "simple-notification",
      "Simple Notification",
      importance: Importance.max,
      priority: Priority.max,
    );

    NotificationDetails notificationDetails = NotificationDetails(
      android: androidNotificationDetails,
    );

    Map<String, dynamic> payload = {
      "type": "sipmle-notification",
    };

    await localNotifications.show(
      0,
      "Başlık",
      "İçerik",
      notificationDetails,
      payload: jsonEncode(payload),
    );
  }

  static Future<void> onNotificationTapped(
    NotificationResponse notificationResponse,
  ) async {
    clg("NotificationService.onNotificationTapped(${notificationResponse.id})");

    if (notificationResponse.payload != null) {
      Map<String, dynamic> payload = jsonDecode(notificationResponse.payload!);

      clg("NotificationService.onNotificationTapped(${notificationResponse.id}) => payload = $payload");
    }
  }
}
```

My SplashView initState codes are below:
```Dart
@override
  void initState() {
    super.initState();

    Future.wait([
      Future.delayed(Duration(seconds: 2)),
      NotificationService.init(),
    ]).then((value) async {
      Get.off(() => HomeView());
      NotificationService.checkNotificationClicked();
    });
  }
```

**Comment by MaikuB** [comment]
Can you elaborate on what's the reason behind why you expected `onDidReceiveBackgroundNotificationResponse` to fire? It is documented to be only for notification actions, which you aren't using so would explain why it's never fired. `getNotificationAppLaunchDetails` is  documented to be meant for when a notification launches the app (i.e. from a terminated state). Did you by chance mix up when to use both of these APIs?

**Comment by mehmetext** [comment]
I know that If I click the notification, `onDidReceiveBackgroundNotificationResponse` should be triggered. Am I wrong? I guess I mixed up them :D Can you explain their purpose or send me a link explaining them?

**Comment by MaikuB** [comment]
I explained earlier that it's for notification *actions* so what you mentioned there is incorrect. If you search the name in the readme then you will find it's only ever mentioned in that context. There are also API docs, see https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/FlutterLocalNotificationsPlugin/initialize.html. Closing this as it's due to a misunderstanding
